packages/core/src/components/Menu/MenuItem/components/BaseMenuItem/BaseMenuItem.tsx

```diff
@@ -0,0 +1,164 @@
+      if (!isActive || !shouldScrollMenu || !anchorElement) {
+        return;
+      }
+      if (anchorElement.scrollIntoViewIfNeeded) {
+        anchorElement.scrollIntoViewIfNeeded({ behaviour: "smooth" });
+        return;
+      }
+      anchorElement.scrollIntoView?.({ behavior: "smooth", block: "center" });
+    }, [isActive, shouldScrollMenu]);
+
+    useEffect(() => {
+      const anchorElement = anchorRef.current;
+      if (useDocumentEventListeners) return;
+      if (isActive) {
+        anchorElement?.focus();
+      }
+    }, [isActive, useDocumentEventListeners]);
+
+    const isMouseEnter = useMenuItemMouseEvents({
+      ref: anchorRef,
+      splitMenuItemIconButtonRef,
+      resetOpenSubMenuIndex,
+      setSubMenuIsOpenByIndex,
+      isActive,
+      setActiveItemIndex,
+      index,
+      hasChildren: Boolean(subMenu),
+      splitMenuItem
+    });
+
+    const { onClickCallback } = useMenuItemKeyboardEvents({
+      onClick,
+      disabled,
+      isActive,
+      index,
+      setActiveItemIndex,
+      hasChildren: Boolean(subMenu),
+      shouldShowSubMenu,
+      setSubMenuIsOpenByIndex,
+      menuRef,
+      isMouseEnter,
+      closeMenu,
+      useDocumentEventListeners,
+      splitMenuItem,
+      isMouseEnterMenuItem,
+      isMouseEnterIconButton
+    });
+
+    const closeSubMenu = useCallback(
+      (option: CloseMenuOption = {}) => {
+        setSubMenuIsOpenByIndex(index, false);
+        if (option.propagate) {
+          closeMenu(option);
+        }
+      },
+      [setSubMenuIsOpenByIndex, index, closeMenu]
+    );
+
+    return (
+      <Text
+        id={id}
+        element="li"
+        type={Text.types.TEXT2}
+        aria-haspopup={subMenu ? true : undefined}
+        aria-expanded={subMenu ? shouldShowSubMenu : undefined}
+        data-testid={dataTestId || getTestId(ComponentDefaultTestId.MENU_ITEM, index)}
+        className={cx(styles.item, className, {
+          [styles.disabled]: disabled,
+          [styles.focused]: isActive,
+          [styles.selected]: selected,
+          [styles.initialSelected]: isInitialSelectedState,
+          [styles.splitMenuItem]: splitMenuItem
+        })}
+        ref={mergedRef}
+        onClick={onClickCallback}
+        role="menuitem"
+        aria-current={isActive}
+        onMouseLeave={onMouseLeave}
+        onMouseEnter={onMouseEnter}
+        tabIndex={TAB_INDEX_FOCUS_WITH_JS_ONLY}
+      >
+        {children}
+        {Boolean(subMenu) && (
+          <>
+            <MenuItemSubMenuIcon
+              ref={splitMenuItemIconButtonRef}
+              isSplit={splitMenuItem}
+              active={shouldShowSubMenu}
+              disabled={disabled}
+            />
+            <MenuItemSubMenu
+              anchorRef={anchorRef}
+              open={shouldShowSubMenu}
+              onClose={closeSubMenu}
+              autoFocusOnMount={!useDocumentEventListeners}
+            >
+              {subMenu}
+            </MenuItemSubMenu>
```

Comment:
Why are these in the Base and not in the MenuItem?

Reply:
cause I want to be able to add sub menu for CustomMenuItem and for each MenuItem variant
MenuItem is an opinionated implemenation now, but the basic should have the capabilities for sub menu

Reply:
Is this intended to replace the option to pass a component to the menu item using the title prop?

Reply:
CustomMenuItem is the one intended to replace, basically yes.
but it puts more weight on the developer to "know what he's doing".
BaseMenuItem is internal

Reply:
Sounds legit to me, but I'm wondering why should devs use the old MenuItem currently and not just the new CustomMenuItem? It gives them more flexibility in the first place and behaves the same

Reply:
they shouldn't basically
only for tooltip reasons (if they want)